delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResolversRoutes.scala

```diff
@@ -0,0 +1,206 @@
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.PaginationConfig
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchParams.ResolverSearchParams
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchResults.{searchResultsEncoder, SearchEncoder}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{BaseUri, IdSegment, Label}
+import io.circe.Json
+import kamon.instrumentation.akka.http.TracingDirectives.operationName
+import monix.execution.Scheduler
+
+/**
+  * The resolver routes
+  * @param identities the identity module
+  * @param acls  the acls module
+  * @param projects the projects module
+  * @param resolvers the resolvers module
+  */
+final class ResolversRoutes(identities: Identities, acls: Acls, projects: Projects, resolvers: Resolvers)(implicit
+    baseUri: BaseUri,
+    paginationConfig: PaginationConfig,
+    s: Scheduler,
+    cr: RemoteContextResolution,
+    ordering: JsonKeyOrdering
+) extends AuthDirectives(identities, acls)
+    with CirceUnmarshalling {
+
+  import baseUri.prefixSegment
+  implicit val resolverContext: ContextValue = Resolvers.context
+
+  private val Read  = resolverPermissions.read
+  private val Write = resolverPermissions.write
+
+  private def resolverSearchParams(project: ProjectRef)(implicit caller: Caller): Directive1[ResolverSearchParams] =
+    searchParams.tflatMap { case (deprecated, rev, createdBy, updatedBy) =>
+      callerAcls.map { aclsCol =>
+        ResolverSearchParams(
+          Some(project),
+          deprecated,
+          rev,
+          createdBy,
+          updatedBy,
+          resolver => aclsCol.exists(caller.identities, Read, AclAddress.Project(resolver.project))
+        )
+      }
+    }
+
+  def routes: Route =
+    baseUriPrefix(baseUri.prefix) {
+      extractCaller { implicit caller =>
+        pathPrefix("resolvers") {
+          concat(
+            // SSE resolvers for all events
+            (pathPrefix("events") & pathEndOrSingleSlash) {
+              get {
+                operationName(s"$prefixSegment/resolvers/events") {
+                  authorizeFor(AclAddress.Root, events.read).apply {
+                    lastEventId { offset =>
+                      emit(resolvers.events(offset))
+                    }
+                  }
+                }
+              }
+            },
+            (projectRef | projectRefFromUuidsLookup(projects)) { ref =>
+              val projectAddress = AclAddress.Project(ref)
+              val authorizeRead  = authorizeFor(projectAddress, Read)
+              val authorizeWrite = authorizeFor(projectAddress, Write)
+              concat(
+                // List resolvers
+                (get & pathEndOrSingleSlash & extractUri & paginated & resolverSearchParams(ref)) {
+                  (uri, pagination, params) =>
+                    operationName(s"$prefixSegment/resolvers/{org}/{project}") {
```

Comment:
you don't need to repeat this directive in both cases. Before the contact you could have:
```scala
(pathEndOrSingleSlash & operationName(s"$prefixSegment/resolvers/{org}/{project}")) {
   concat(...)
}
```